src/posthog-surveys.ts

```diff
@@ -155,4 +161,101 @@ export class PostHogSurveys {
             return callback(targetingMatchedSurveys)
         }, forceReload)
     }
+
+    getNextSurveyStep(survey: Survey, currentQuestionIndex: number, response: string | string[] | number | null) {
+        const question = survey.questions[currentQuestionIndex]
+        const nextQuestionIndex = currentQuestionIndex + 1
+
+        if (!question.branching?.type) {
+            if (currentQuestionIndex === survey.questions.length - 1) {
+                return SurveyQuestionBranchingType.ConfirmationMessage
+            }
+
+            return nextQuestionIndex
+        }
+
+        if (question.branching.type === SurveyQuestionBranchingType.ConfirmationMessage) {
+            return SurveyQuestionBranchingType.ConfirmationMessage
+        } else if (question.branching.type === SurveyQuestionBranchingType.SpecificQuestion) {
+            if (Number.isInteger(question.branching.index)) {
+                return question.branching.index
+            }
+        } else if (question.branching.type === SurveyQuestionBranchingType.ResponseBased) {
+            // Single choice
+            if (question.type === SurveyQuestionType.SingleChoice) {
+                // :KLUDGE: for now, look up the choiceIndex based on the response
+                // TODO: once QuestionTypes.MultipleChoiceQuestion is refactored, pass the selected choiceIndex into this method
```

Comment:
I think now this is independent of any refactors?

Reminder that choices can be shuffled too, would be nice to see some tests for the single choice case + where the choices are shuffled

Reply:
Should've been clearer - I want to refactor `MultipleChoiceQuestion ` into `SingleSelectQuestion` and `MultipleSelectQuestion`. When that's done, I'll pass `choiceIndex` into this method directly rather than looking it up with the choice value. Want to do it later to not interfere with Dylan (who's also working on this file).

Added the test for single choice w/ shuffling 👍